tracing/src/main/java/com/palantir/tracing/Tracers.java

```diff
@@ -507,34 +507,42 @@ private static void restoreTrace(Optional<Trace> trace) {
      */
     private static class TracingAwareCallable<V> implements Callable<V> {
         private final Callable<V> delegate;
-        private final DeferredTracer deferredTracer;
+        private final DetachedSpan detachedSpan;
+        private final String operation;
+        private final Map<String, String> metadata;
 
         TracingAwareCallable(String operation, Map<String, String> metadata, Callable<V> delegate) {
             this.delegate = delegate;
-            this.deferredTracer = new DeferredTracer(operation, metadata);
+            this.detachedSpan = DetachedSpan.detach();
+            this.operation = operation;
+            this.metadata = metadata;
         }
 
         @Override
         public V call() throws Exception {
-            try (DeferredTracer.CloseableTrace ignored = deferredTracer.withTrace()) {
+            try (CloseableSpan ignored = detachedSpan.childSpan(operation, metadata)) {
                 return delegate.call();
+            } finally {
+                detachedSpan.complete();
```

Comment:
These `complete` calls are entirely unnecessary, however the reuse of `DetachedSpan` suggests we should use them in case we're given a 'real' detached span -- also for copy/pasted code.